src/swaps/lifi/LifiSwapProvider.ts

```diff
@@ -0,0 +1,321 @@
+    const updates = approveTx || (await this.initiateSwap(swap));
+
+    return { id: uuidv4(), ...updates };
+  }
+
+  private async initiateSwap({ network, walletId, quote }: SwapRequest<LifiSwapHistoryItem>) {
+    const client = super.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const txData: TransactionRequest = quote.bestRoute?.transactionRequest as TransactionRequest;
+    const fromFundTx = await client.wallet.sendTransaction(txData);
+
+    return {
+      status: 'WAITING_FOR_INITIATION_CONFIRMATIONS',
+      fromFundTx,
+      fromFundHash: fromFundTx.hash,
+    };
+  }
+
+  //  ======== FEES ========
+  async estimateFees(feeRequest: EstimateFeeRequest<string, LifiSwapHistoryItem>) {
+    if (feeRequest.txType in this._txTypes()) {
+      const fees: EstimateFeeResponse = {};
+      const gasCost = (feeRequest.quote.bestRoute?.estimate.gasCosts as GasCost[])[0];
+
+      const approvalData = await this.buildApprovalTx(
+        feeRequest,
+        false,
+        feeRequest.quote.bestRoute?.estimate.approvalAddress
+      );
+
+      let approvalGas = 0;
+      if (approvalData) {
+        const client = this.getClient(
+          feeRequest.network,
+          feeRequest.walletId,
+          feeRequest.quote.from,
+          feeRequest.quote.fromAccountId
+        );
+        approvalGas = (await client.chain.getProvider().estimateGas(approvalData)).toNumber();
+      }
+
+      const nativeAsset = chains[cryptoassets[feeRequest.quote.from].chain].nativeAsset;
+
+      // have only one fee and it cannot be changed by user
+      for (const feePrice of feeRequest.feePrices) {
+        const gasPrice = new BN(feePrice).times(1e9); // ETH fee price is in gwei
+        const approvalFee = new BN(approvalGas).times(gasPrice);
+        fees[feePrice] = unitToCurrency(cryptoassets[nativeAsset], approvalFee.plus(gasCost.amount));
+      }
+
+      return fees;
+    }
+
+    return null;
+  }
+
+  async getMin(_quoteRequest: QuoteRequest) {
+    return new BN(0);
+  }
+
+  // ======== HELPERS ========
+
+  private getDefaultRouteOptions() {
+    return {
+      slippage: 5 / 100, // 5%
+      order: Orders[0], // 'RECOMMENDED'
+    };
+  }
+
+  // cross chain swaps info can be aquire only via Li.Fi API
+  private async getCrossChainSwapStatus(quote: LifiSwapHistoryItem) {
+    const result = await this._httpClient.nodeGet('/status', {
+      bridge: quote.bestRoute?.tool,
+      fromChain: this.getChainByID(quote.bestRoute?.action.fromChainId as number),
+      toChain: this.getChainByID(quote.bestRoute?.action.toChainId as number),
+      txHash: quote.fromFundHash,
+    });
+
+    return result;
+  }
+
+  private getChainByID(id: number) {
+    const indexOfFromChainId = Object.values(ChainId).indexOf(id as unknown as ChainId);
+    return Object.keys(ChainId)[indexOfFromChainId];
+  }
+
+  // lifi type of swaps are simillar to boost (combination of multiple single and/or cross chain swaps)
```

Comment:
In this case, are all lifi swaps only 1 wallet transaction? worth checking...

Reply:
What do you mean by 1 wallet? Especially when we talk about Liquality wallet?
There are 3 types of swaps:
1) 'Swap' -> single chain (like uniswap)
2) 'Cross' -> multi chain (bridging like hop for example)
3) 'LiFi' -> some combination of the previous 2 types (you can have 2 swaps or 3 swaps and 1 cross )

Reply:
I mean if they are 1 transaction. I  am wondering if some of the lifi swaps might require more than 1 transaction from the user and if our provider is supporting that correctly.